src/codeu/chat/server/Model.java

```diff
@@ -134,6 +142,24 @@ public void add(Message message) {
   public StoreAccessor<String, Message> messageByText() {
     return messageByText;
   }
+<<<<<<< HEAD
+
+  public void changeAuthority(ConversationHeader conversation, Uuid targetUser, byte authorityByte)
+  {
+    LinkedHashMap<Uuid,Byte> userAuthority;
+    if(authority.get(conversation.id) == null)
+    {
+       authority.put(conversation.id, new LinkedHashMap<>());
+    }
+    userAuthority = authority.get(conversation.id);
+    userAuthority.put(targetUser, authorityByte);
+  }
+
+  public boolean validateAuthority(ConversationHeader conversation, Uuid targetUser, byte authorityByte)
+  {
+    LinkedHashMap<Uuid, Byte> userAuthority = authority.get(conversation.id);
+    return userAuthority.get(conversation.id) != null && userAuthority.get(targetUser).byteValue() == authorityByte;
```

Comment:
This isn't quite right - remember that the user doesn't need to exactly match the required `authorityByte` - if they have a superset of the required authority bits, they should still be allowed to perform the action in question. (If the user has bits `011`, but the `authorityByte` argument is `001`, this method should still return true.

You can use [bitwise operations](https://docs.oracle.com/javase/tutorial/java/nutsandbolts/op3.html) to do this in a clever way!